### PR TITLE
Create Title ui component and update TabPanelitem and BaseButton components

### PR DESCRIPTION
**Pull request by ckrac**
- create Chip and Title ui component
- add 'auto', string and number union type to width prop of  BaseButton
- add cursorPointer prop to BaseButton to change css cursor property value to pointer
- update TabPanelItem to use Title ui component to render section title and add optional TitleTag prop to render in Title component

### CODE REVIEW

**Review by asucarlos — APPROVED** [review]

File: src/ui/src/components/TabPanelItem/TabPanelItem.tsx [hunk 1]
```diff
@@ -23,12 +25,20 @@ const TabPanelItem: FC<TabPanelItemProps> = ({
     'no-padding-bottom': noPaddingBotttom,
     'little-padding-y': littlePaddingY,
   })
+
+  const showTitleUi = !!title || !!TitleTag
```
Comment (kowsheek) [inline src/ui/src/components/TabPanelItem/TabPanelItem.tsx:29]: Can this logic be simpler?

File: src/ui/src/components/BaseButton/BaseButton.tsx [hunk 2]
```diff
@@ -22,7 +24,11 @@ const BaseButton: FC<BaseButtonProps> = ({ width, fullWidth, style, onClick, ...
   return (
     <button
       {...props}
-      className={`btn ${props.className}`}
+      className={cn('btn', {
+        [props.className || '']: !!props.className,
```
Comment (kowsheek) [inline src/ui/src/components/BaseButton/BaseButton.tsx:28]: What does this do?

**Review by kowsheek — CHANGES_REQUESTED** [review]

File: src/ui/src/components/TabPanelItem/TabPanelItem.tsx (hunk 1 above)
Comment (ckrac) [reply src/ui/src/components/TabPanelItem/TabPanelItem.tsx:29]: I guess I can update it to only show the title if title prop is passed.

File: src/ui/src/components/BaseButton/BaseButton.tsx (hunk 2 above)
Comment (ckrac) [reply src/ui/src/components/BaseButton/BaseButton.tsx:28]: cn function can take an object and the key in the object is the class string that can be added if it's value is true. In this case props.className might be undefined so typescript will show an error and to fix this, the `|| ''` is added.

**Pull request merged**